zebrad/src/components/mempool/gossip.rs

```diff
@@ -37,14 +37,35 @@ where
     let mut broadcast_network = Timeout::new(broadcast_network, TIPS_RESPONSE_TIMEOUT);
 
     loop {
-        // once we get new data in the channel, broadcast to peers
+        let mut combined_changes = 1;
+
+        // once we get new data in the channel, broadcast to peers,
+        // the mempool automatically combines some transactions that arrive close together
         receiver.changed().await?;
+        let mut txs = receiver.borrow().clone();
+        tokio::task::yield_now().await;
+
+        // also combine transactions that arrived shortly after this one
+        while receiver.has_changed()? && combined_changes < MAX_CHANGES_BEFORE_SEND {
+            // Correctness: reset has_changed(), don't hold the lock
```

Comment:
This comment is not totally clear to me. When we are resetting and what ?

Reply:
I expanded it in commit 09912976983923ec59dcecb366c0979a98ffa981. The relevant API docs are:
https://docs.rs/tokio/latest/tokio/sync/watch/struct.Receiver.html#method.borrow_and_update
https://docs.rs/tokio/latest/tokio/sync/watch/struct.Receiver.html#method.has_changed